zingolib/src/wallet/transaction_records_by_id.rs

```diff
@@ -39,16 +49,252 @@ impl TransactionRecordsById {
     pub fn new() -> Self {
         TransactionRecordsById(HashMap::new())
     }
-    // Constructs a TransactionRecordMap from a HashMap
+    /// Constructs a TransactionRecordsById from a HashMap
     pub fn from_map(map: HashMap<TxId, TransactionRecord>) -> Self {
         TransactionRecordsById(map)
     }
+    pub fn get_received_note_from_identifier<D: DomainWalletExt>(
+        &self,
+        note_record_reference: NoteRecordIdentifier,
+    ) -> Option<
+        zcash_client_backend::wallet::ReceivedNote<
+            NoteRecordIdentifier,
+            <D as zcash_note_encryption::Domain>::Note,
+        >,
+    >
+    where
+        <D as zcash_note_encryption::Domain>::Note: PartialEq + Clone,
+        <D as zcash_note_encryption::Domain>::Recipient: super::traits::Recipient,
+    {
+        let transaction = self.get(&note_record_reference.txid);
+        transaction.and_then(|transaction_record| {
```

Comment:
This seems to treat a found transparent note in the same way as no note.  Is that correct?!

Reply:
In both cases this function returns `None`.